src/renderer/components/SketchParams/SketchParams.tsx

```diff
@@ -30,8 +32,11 @@ const getInputElement = (valueType: NodeTypes, id: string) => {
 }
 
 const ParamItem = ({ param: { key, title, id, valueType } }: ParamProps) => {
+  //const onSelectNode = useOnSelectNode(id)
+  const onSelectNode = useEngineStore((state) => state.updateSelectedNode).bind(null, id)
```

Comment:
This use of `.bind` comes with a small performance cost. `.bind` creates a function every time it is called. So whenever `ParamItem` is called, `onSelectNode` will be a new value every time. This gets passed down as a prop and `NodeControl` will need re-rendering every time, because it will evaluate it as a new update to the component. It's probably minuscule in this example, but over time these sorts of things can build up and create little frame skips etc. 

This is the reason why we use something like `useCallback` because it takes a dependency array. Only if the items of the array change will it create a new function. And because of all that annoying stuff, I create a hook to hide it all away from the component 😉